db/models/api.model.js

```diff
@@ -1,18 +1,46 @@
-const db = require('../connection')
+const db = require("../connection");
 
 function readTopics() {
-    return db.query('SELECT * FROM topics;').then(({rows}) => {
-        return rows
-    })
+  return db.query("SELECT * FROM topics;").then(({ rows }) => {
+    return rows;
+  });
 }
 
 function readArticleId(articleId) {
-   // console.log(articleId)
- return db.query('SELECT * FROM articles WHERE article_id = $1;', [articleId])
- .then(({rows}) => {
-    return rows[0];
- })
+  // console.log(articleId)
+  return db
+    .query("SELECT * FROM articles WHERE article_id = $1;", [articleId])
+    .then(({ rows }) => {
+      return rows[0];
+    });
 }
 
+function readArticles() {
+  let sqlQuery = `
+SELECT 
+      articles.article_id,
+      articles.title,
+      articles.topic,
+      articles.author,
+      articles.created_at,
+      articles.votes,
+      articles.article_img_url,
+    COALESCE(COUNT(comments.article_id), 0) AS comment_count
+FROM articles
+LEFT JOIN comments
+ON articles.article_id = comments.article_id
+    GROUP BY articles.article_id
+    ORDER BY articles.created_at DESC;
+`;
 
-module.exports = {readTopics, readArticleId}
\ No newline at end of file
+  const queryValues = [];
+
+  return db.query(sqlQuery, queryValues).then(({ rows }) => {
+    rows.forEach((article) => {
+      article.comment_count = parseInt(article.comment_count);
+    });
```

Comment:
If you want to alter the comment_count data type to be a number it is possible to do it in the query above. Have a look into casting. It will be a bit neater than doing it with js (although it is not wrong!)